static/app/views/alerts/rules/metric/presets.tsx

```diff
@@ -1,269 +1,204 @@
-import type {LinkProps} from 'sentry/components/links/link';
+import {Client} from 'sentry/api';
+import {IconBookmark, IconGlobe, IconGraph} from 'sentry/icons';
 import {t} from 'sentry/locale';
-import {Project} from 'sentry/types';
-import {DisplayModes} from 'sentry/utils/discover/types';
-import {MutableSearch} from 'sentry/utils/tokenizeSearch';
-import {Incident, IncidentStats} from 'sentry/views/alerts/types';
-import {getStartEndFromStats} from 'sentry/views/alerts/utils';
-import {getIncidentDiscoverUrl} from 'sentry/views/alerts/utils/getIncidentDiscoverUrl';
-import {transactionSummaryRouteWithQuery} from 'sentry/views/performance/transactionSummary/utils';
-
-import {Dataset} from './types';
+import {Organization, Project} from 'sentry/types';
+import {uniqueId} from 'sentry/utils/guid';
+
+import {AlertType} from '../../wizard/options';
+
+import {
+  ActionType,
+  AlertRuleComparisonType,
+  AlertRuleThresholdType,
+  AlertRuleTriggerType,
+  Dataset,
+  EventTypes,
+  TargetType,
+  Trigger,
+} from './types';
+
+export type PresetContext = {
+  aggregate: string;
+  comparisonType: AlertRuleComparisonType;
+  dataset: Dataset;
+  eventTypes: EventTypes[];
+  name: string;
+  thresholdType: AlertRuleThresholdType;
 
-type PresetCta = {
-  /**
-   * The CTA text
-   */
-  buttonText: string;
-  /**
-   * The location to direct to upon clicking the CTA.
-   */
-  to: LinkProps['to'];
-  /**
-   * The tooltip title for the CTA button, may be empty.
-   */
-  title?: string;
+  triggers: Trigger[];
+  comparisonDelta?: number;
+  query?: string;
+  timeWindow?: number;
 };
-
-type PresetCtaOpts = {
-  orgSlug: string;
-  projects: Project[];
-  incident?: Incident;
-  stats?: IncidentStats;
+export type Preset = {
+  Icon: typeof IconGlobe | typeof IconGraph | typeof IconBookmark;
+  // Will be shown on the corresponding alert type in the wizard.
+  alertType: AlertType;
+  description: string;
+  id: string;
+  makeContext(
+    client: Client,
+    project: Project,
+    organization: Organization
+  ): Promise<PresetContext>;
+  title: string;
 };
 
-type Preset = {
-  /**
-   * The default aggregate to use when selecting this preset
-   */
-  default: string;
-  /**
-   * Generates the CTA component
-   */
-  makeCtaParams: (opts: PresetCtaOpts) => PresetCta;
-  /**
-   * The regex used to match aggregates to this preset.
-   */
-  match: RegExp;
-  /**
-   * The name of the preset
-   */
-  name: string;
-  /**
-   * The dataset that this preset applys to.
-   */
-  validDataset: Dataset[];
-};
+async function getHighestVolumeTransaction(
+  client: Client,
+  organizationSlug: string,
+  projectId: string
+): Promise<[string, number] | null> {
+  const result = await client.requestPromise(
+    `/organizations/${organizationSlug}/events/`,
+    {
+      method: 'GET',
+      data: {
+        statsPeriod: '30d',
```

Comment:
@Neo-Zhixing Can we just use the last 7 days?

Reply:
Changed to `7d`